drivers/block-nbd.c

```diff
@@ -680,6 +680,7 @@ tdnbd_connect_import_session(struct tdnbd_data *prv, td_driver_t* driver)
 			sizeof(opt));
 	if (rc < 0) {
 		ERROR("Could not set TCP_NODELAY: %s\n", strerror(errno));
+		close(sock);
 		return -1;
 	}
```

Comment:
Not your code, but do we need to free the prv->remote on line 716

Reply:
Looks like it. Will check why we don't get a resource leak event there.

Reply:
Done